File: chad/src/main/java/org/woahoverflow/chad/commands/fun/PhotoEditor.java
package org.woahoverflow.chad.commands.fun;

import org.jetbrains.annotations.NotNull;
import org.woahoverflow.chad.framework.handle.GuildHandler;
import org.woahoverflow.chad.framework.handle.JsonHandler;
import org.woahoverflow.chad.framework.handle.MessageHandler;
import org.woahoverflow.chad.framework.obj.Command;
import org.woahoverflow.chad.framework.obj.Guild;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent;
import sx.blah.discord.util.EmbedBuilder;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.HashMap;
import java.util.List;
import java.util.Objects;

/**
 * Use preset options to modify a photo
 *
 * @author sho, codebasepw
 */
public class PhotoEditor implements Command.Class {
    @Override
    public final Runnable run(@NotNull MessageEvent e, @NotNull List<String> args) {
        return () -> {
            MessageHandler messageHandler = new MessageHandler(e.getChannel(), e.getAuthor());
            String prefix = (String) GuildHandler.getGuild(e.getGuild().getLongID()).getObject(Guild.DataType.PREFIX);

            // Makes sure the user has attached a file
            if (e.getMessage().getAttachments().isEmpty()) {
                messageHandler.sendError("No file was found!");
                return;
            }

            // Makes sure they added arguments
            if (args.isEmpty()) {
                messageHandler.sendPresetError(MessageHandler.Messages.INVALID_ARGUMENTS, prefix + "pe **deepfry**");
                return;
            }

            // Makes sure the attachment is a PNG or JPG
            if (!(e.getMessage().getAttachments().get(0).getUrl().endsWith(".png") || e.getMessage().getAttachments().get(0).getUrl().endsWith(".jpg"))) {
                new MessageHandler(e.getChannel(), e.getAuthor()).sendError("Invalid Format!\nPlease use PNG or JPG");
                return;
            }

            // Assigns the URL to the attachment's URL
            URL url;
            try {
                url = new URL(e.getMessage().getAttachments().get(0).getUrl());
            } catch (MalformedURLException e1) {
                messageHandler.sendPresetError(MessageHandler.Messages.INTERNAL_EXCEPTION);
                return;
            }


            // Deepfry
            if (args.get(0).equalsIgnoreCase("deepfry")) {
                messageHandler.sendEmbed(new EmbedBuilder().withImage(
                        Objects.requireNonNull(JsonHandler.INSTANCE.read("https://nekobot.xyz/api/imagegen?type=deepfry&image=" + url)).getString("message"))
                );
                return;
            }

            // If none of the arguments were met, return;
            messageHandler.sendPresetError(MessageHandler.Messages.INVALID_ARGUMENTS, prefix + "pe **deepfry**");
        };
    }

    @Override
    public final Runnable help(@NotNull MessageEvent e) {
        HashMap<String, String> st = new HashMap<>();
        st.put("pe deepfry <image>", "Deepfries an image.");
        return Command.helpCommand(st, "Photo Editor", e);
    }
}

File: chad/src/main/java/org/woahoverflow/chad/commands/fun/PhotoEditor.kt
package org.woahoverflow.chad.commands.`fun`

import org.json.JSONObject
import org.woahoverflow.chad.framework.handle.GuildHandler
import org.woahoverflow.chad.framework.handle.JsonHandler
import org.woahoverflow.chad.framework.handle.MessageHandler
import org.woahoverflow.chad.framework.obj.Command
import org.woahoverflow.chad.framework.obj.Guild
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder

import java.net.MalformedURLException
import java.net.URL
import java.util.HashMap
import java.util.Objects

/**
 * Use preset options to modify a photo
 *
 * @author sho, codebasepw
 */
class PhotoEditor : Command.Class {
    override suspend fun run(e: MessageEvent, args: MutableList<String>) {
        val messageHandler = MessageHandler(e.channel, e.author)
        val prefix = GuildHandler.getGuild(e.guild.longID).getObject(Guild.DataType.PREFIX) as String

        // Makes sure the user has attached a file
        if (e.message.attachments.isEmpty()) {
            messageHandler.sendError("No file was found!")
            return
        }

        // Makes sure they added arguments
        if (args.isEmpty()) {
            messageHandler.sendPresetError(MessageHandler.Messages.INVALID_ARGUMENTS, prefix + "pe [deepfry]")
            return
        }

        // Makes sure the attachment is a PNG or JPG
        if (!(e.message.attachments[0].url.endsWith(".png") || e.message.attachments[0].url.endsWith(".jpg"))) {
            MessageHandler(e.channel, e.author).sendError("Invalid Format!\nPlease use PNG or JPG")
            return
        }

        // Assigns the URL to the attachment's URL
        val url: URL
        try {
            url = URL(e.message.attachments[0].url)
        } catch (e1: MalformedURLException) {
            messageHandler.sendPresetError(MessageHandler.Messages.INTERNAL_EXCEPTION)
            return
        }


        // Deepfry
        if (args[0].equals("deepfry", ignoreCase = true)) {
            messageHandler.sendEmbed(EmbedBuilder().withImage(
                    Objects.requireNonNull<JSONObject>(JsonHandler.read("https://nekobot.xyz/api/imagegen?type=deepfry&image=$url")).getString("message"))
            )
            return
        }

        // If none of the arguments were met, return;
        messageHandler.sendPresetError(MessageHandler.Messages.INVALID_ARGUMENTS, prefix + "pe [deepfry]")
    }

    override suspend fun help(e: MessageEvent) {
        val st = HashMap<String, String>()
        st["pe deepfry [image]"] = "Deepfries an image."
        Command.helpCommand(st, "Photo Editor", e)
    }
}
